pkg/providers/aws/provider_redis.go

```diff
@@ -168,6 +173,21 @@ func (p *AWSRedisProvider) createElasticacheCluster(ctx context.Context, r *v1al
 		return nil, croType.StatusMessage(fmt.Sprintf("createReplicationGroup() in progress, current aws elasticache status is %s", *foundCache.Status)), nil
 	}
 
+	namespace := r.Namespace
+	resourceID := r.Name
+	instanceID := *foundCache.ReplicationGroupId
+	clusterID, err := resources.GetClusterID(ctx, p.Client)
+	if err != nil {
+		errMsg := "failed to retrieve cluster identifier"
+		return nil, croType.StatusMessage(errMsg), errorUtil.Wrap(err, errMsg)
+	}
+	// Create the PrometheusRule alert to watch for the availability of this ElastiCache instance we are provisioning
+	err = p.CreateElastiCacheAvailabilityAlert(ctx, resourceID, namespace, instanceID, clusterID)
```

Comment:
It might be easier to pass the `Redis` type directly, or are we keeping it as two strings to keep this function generic and re-use for an RDS availability alert

Reply:
In this case we can just pass the CR yeah, this function is unique to creating the availability alerts for redis.